kvx/src/implementations/postgres.rs

```diff
@@ -85,57 +86,50 @@ impl<E: HasExecutor> KeyValueStoreBackend for Postgres<E> {
 
 impl<E: HasExecutor> ReadStore for Postgres<E> {
     fn has(&self, key: &Key) -> Result<bool> {
-        let key = key.with_namespace(self.namespace.clone());
-
         Ok(self
             .executor
             .executor()?
             .exec_query_opt(
-                "SELECT 1 FROM store WHERE scope = $1 AND key = $2",
-                &[key.scope().as_vec(), &key.name()],
+                "SELECT 1 FROM store WHERE namespace = $1 AND scope = $2 AND key = $3",
+                &[&self.namespace, key.scope().as_vec(), &key.name()],
             )?
             .is_some())
     }
 
     fn has_scope(&self, scope: &Scope) -> Result<bool> {
-        let scope = scope.with_namespace(self.namespace.clone());
-
         Ok(self
             .executor
             .executor()?
             .exec_query_opt(
-                "SELECT 1 FROM store WHERE scope[:$2]  = $1",
-                &[scope.as_vec(), &scope.len()],
+                "SELECT 1 FROM store WHERE namespace = $1 AND scope[:$3]  = $2",
+                &[&self.namespace, scope.as_vec(), &scope.len()],
             )?
             .is_some())
     }
 
     fn get(&self, key: &Key) -> Result<Option<serde_json::Value>> {
-        let key = key.with_namespace(self.namespace.clone());
         Ok(self
             .executor
             .executor()?
             .exec_query_opt(
-                "SELECT value FROM store WHERE scope = $1 AND key = $2",
-                &[key.scope().as_vec(), &key.name()],
+                "SELECT value FROM store WHERE namespace = $1 AND scope = $2 AND key = $3",
+                &[&self.namespace, key.scope().as_vec(), &key.name()],
             )?
             .and_then(|row| row.get(0)))
     }
 
     fn list_keys(&self, scope: &Scope) -> Result<Vec<Key>> {
-        let scope = scope.with_namespace(self.namespace.clone());
         Ok(self
             .executor
             .executor()?
             .exec_query(
-                "SELECT scope, key FROM store WHERE scope[:$2] = $1",
-                &[scope.as_vec(), &scope.len()],
+                "SELECT scope, key FROM store WHERE namespace = $1 AND scope[:$3] = $2",
+                &[&self.namespace, scope.as_vec(), &scope.len()],
             )?
             .into_iter()
             .map(|row| {
-                let scope: Vec<SegmentBuf> = row.get(0);
-                let mut scope = Scope::new(scope);
-                scope.remove_namespace(self.namespace.clone());
+                let scope_segments: Vec<SegmentBuf> = row.get(0);
```

Comment:
I think you can merge these two lines into a simpl `Scope::new(row.get(0))`.

Reply:
done